test/pokemon-router-test.js

```diff
@@ -0,0 +1,127 @@
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('Charmander');
+          expect(res.body.type).to.equal('fire');
+          expect(res.body.pokedexNUM).to.equal(4);
+          expect(Boolean(res.body.id)).to.equal(true);
+          done();
+        })
+        .catch(done);
+      });
+    });
+    describe('with invalid input or not found input', function(){
+      it('should respond with bad request', done => {
+        superagent.post(`${apiURL}/api/pokemon`)
+        .send({})
+        .then(done)
+        .catch(err => {
+          expect(err.status).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('testing GET', function() {
+    describe('with valid input', function() {
+      before(done => {
+        this.tempPokemon = new Pokemon({
+          name:'Charmander',
+          type: 'fire',
+          pokedexNUM: 4,
+          moves: '"scratch", "ember", "tail whip"',
+        });
+        storage.setItem('pokemon', this.tempPokemon)
+        .then(() => done())
+        .catch(done);
+      });
+
+      it('should return a pokemon from a given id', (done) => {
+        superagent.get(`${apiURL}/api/pokemon?id=${this.tempPokemon.id}`)
+        .then(res => {
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('Charmander');
+          expect(res.body.type).to.equal('fire');
+          expect(res.body.pokedexNUM).to.equal(4);
+          expect(Boolean(res.body.id)).to.equal(true);
+          done();
+        })
+        .catch(done);
+      });
+    });
+
+    describe('with invalid input', function() {
+      before(done => {
+        this.tempPokemon = new Pokemon({
+          name:'Charmander',
+          type: 'fire',
+          pokedexNUM: 4,
+          moves: '"scratch", "ember", "tail whip"',
+        });
+        storage.setItem('pokemon', this.tempPokemon)
+        .then(() => done())
+        .catch(done);
+      });
+      it('no id provided should respond bad request', done => {
+        superagent.get(`${apiURL}/api/pokemon?id=`)
+        .then(done)
+        .catch(err => {
+          expect(err.status).to.equal(400);
+          done();
+        });
+      });
+      it('id not found should respond with not found', done => {
+        superagent.get(`${apiURL}/api/pokemon?id=5556666`)
+        .then(done)
+        .catch(err => {
+          expect(err.status).to.equal(404);
+          done();
+        });
+      });
+    });
+  });
+  describe('testing DELETE', function() {
+    describe('with valid input', function() {
+      before(done => {
+        this.tempPokemon = new Pokemon({
+          name:'Charmander',
+          type: 'fire',
+          pokedexNUM: 4,
+          moves: '"scratch", "ember", "tail whip"',
+        });
+        storage.setItem('pokemon', this.tempPokemon)
+        .then(() => done())
+        .catch(done);
+      });
+
+      it('should delete a pokemon from a given id', (done) => {
```

Comment:
Great job testing your DELETE route and getting it to pass!

Your expect statement is fine here, but you can add a more robust expect statement for a DELETE test:

```
expect(res.status).to.equal(204);
```